db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/DeleteNode.java

```diff
@@ -112,6 +114,32 @@ public void init(Object targetTableName,
         this.targetProperties = (Properties) targetProperties;
     }
 
+    static public boolean isBulkDelete(Properties properties)
+    {
+        return properties == null ? false :
+                properties.getProperty(BULK_DELETE_DIRECTORY) != null;
+    }
+
+    static public Properties getDeleteProperties(Properties tableProperties)
+    {
+        if (tableProperties == null) {
+            return null;
+        }
+        String dir = (String)tableProperties.remove(BULK_DELETE_DIRECTORY);
```

Comment:
Out of curiosity, why do we call `remove` here causing a side-effect? If that's intentional, should be maybe change the method name into something like `spliceDeleteProperties` ( no puns intended! :) )

Reply:
you are right. i have no idea, i just mimicked the original code